TicTacToe2/TicTacToeGame.cs

```diff
@@ -136,11 +136,17 @@ private bool CheckForDraw()
         }
 
         private bool CheckForWin()
+        {
+            if (CheckHorizontal() == true || CheckVertical() == true || CheckDiagonals() == true)
+            {
+                return true;
+            }
+            return false;
```

Comment:
This logic can be reduced to
```c#
return CheckHorizontal() || CheckVertical() || CheckDiagonals();
```

But my preference is to move the logic for printing winner out of the sub-functions, so I would go with something like this.

```c#
bool win = CheckHorizontal() || CheckVertical() || CheckDiagonals();

if (win)
{
    Winner();
}

return win;
```